src/v/cloud_storage/partition_manifest.cc

```diff
@@ -267,12 +307,76 @@ bool partition_manifest::contains(const segment_name& name) const {
     return _segments.contains(key);
 }
 
+void partition_manifest::delete_replaced_segments() { _replaced.clear(); }
+
+bool partition_manifest::advance_start_offset(model::offset new_start_offset) {
+    if (new_start_offset > _start_offset && !_segments.empty()) {
+        key k = {
+          .base_offset = new_start_offset,
+          .term = model::term_id::max(),
+        };
+        auto it = _segments.upper_bound(k);
+        if (it == _segments.begin()) {
+            return false;
+        }
+        it = std::prev(it);
+        if (it->second.committed_offset < new_start_offset) {
+            auto n = std::next(it);
+            if (n == _segments.end()) {
+                // The whole offset range is supposed to be truncated
+                _start_offset = new_start_offset;
+            } else {
+                _start_offset = n->second.base_offset;
+            }
+        } else {
+            _start_offset = it->second.base_offset;
+        }
+        return true;
+    }
+    return false;
+}
+
+std::vector<segment_meta> partition_manifest::replaced_segments() const {
+    std::vector<segment_meta> segments;
+    segments.reserve(_replaced.size());
+    for (const auto& kv : _replaced) {
+        segments.push_back(kv.second);
+    }
+    return segments;
+}
+
+void partition_manifest::move_aligned_offset_range(
```

Comment:
I don't like the new name either :) (some q's while looking at it: where do we move the offset range and what is it aligned to?) But anyway, definitely a nit and not a blocker.